extensions/wikia/ArticleVideo/bower_components/html5-skin/js/views/pauseScreen.js

```diff
@@ -0,0 +1,148 @@
+/********************************************************************
+ PAUSE SCREEN
+ *********************************************************************/
+var React = require('react'),
+    ReactDOM = require('react-dom'),
+    ClassNames = require('classnames'),
+    ControlBar = require('../components/controlBar'),
+    AdOverlay = require('../components/adOverlay'),
+    UpNextPanel = require('../components/upNextPanel'),
+    TextTrack = require('../components/textTrackPanel'),
+    Watermark = require('../components/watermark'),
+    ResizeMixin = require('../mixins/resizeMixin'),
+    Icon = require('../components/icon'),
+    Utils = require('../components/utils'),
+    AnimateMixin = require('../mixins/animateMixin');
+
+var PauseScreen = React.createClass({
+  mixins: [ResizeMixin, AnimateMixin],
+
+  getInitialState: function() {
+    return {
+      descriptionText: this.props.contentTree.description,
+      controlBarVisible: true
+    };
+  },
+
+  componentDidMount: function() {
+    this.handleResize();
+  },
+
+  componentWillUnmount: function() {
+    this.props.controller.enablePauseAnimation();
+  },
+
+  handleResize: function() {
+    if (ReactDOM.findDOMNode(this.refs.description)){
+      this.setState({
+        descriptionText: Utils.truncateTextToWidth(ReactDOM.findDOMNode(this.refs.description), this.props.contentTree.description)
+      });
+    }
+  },
+
+  handleClick: function(event) {
+    event.preventDefault();
+    this.props.controller.togglePlayPause();
+    this.props.controller.state.accessibilityControlsEnabled = true;
+  },
+
+  render: function() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Function has a complexity of 11 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)